roles/common/defaults/main.yml

```diff
@@ -11,8 +11,9 @@ ats_home: "{{ binaries_folder }}/transform-service"
 
 # Due to https://github.com/ansible/ansible/issues/64957
 # Using default() filter for hostvars 1st level key requires ansible 2.10+
-nginx_host: "{% if groups.nginx | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.nginx[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
-repo_host: "{% if groups.repository | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.repository[0]].ansible_default_ipv4.address | default('127.0.0.1') }}{% endif %}"
+nginx_host: "{{ groups['nginx'] | map('extract', hostvars, ['ansible_default_ipv4','address']) | first | default('127.0.0.1') }}"
+repo_hosts: "{{ groups['repository'] | map('extract', hostvars, ['ansible_default_ipv4','address']) | list }}"
+repo_host: "{% if repo_hosts %}{{ repo_hosts[0] }}{% else %}{{ ['127.0.0.1'] | list }}{% endif %}"
```

Comment:
does it really make sense to provide a 127.0.0.1 default both here and for nginx_host? is there an use case?